oxidation/parsec_cli/parsec_sequestre_dump/src/model.rs

```diff
@@ -0,0 +1,519 @@
+                mut blocks,
+                ..
+            }) => {
+                blocks.sort_by(|x, y| x.offset.cmp(&y.offset));
+
+                let content = blocks
+                    .iter()
+                    .map(|x| {
+                        block::table
+                            .select(block::data)
+                            .filter(block::block_id.eq((*x.id).as_ref()))
+                            .first::<Vec<u8>>(conn)
+                            .map(|x| String::from_utf8_lossy(&x).to_string())
+                    })
+                    .collect::<Result<_, _>>()?;
+
+                Self {
+                    name,
+                    author,
+                    timestamp: *timestamp.as_ref(),
+                    id: *id,
+                    version,
+                    created: *created.as_ref(),
+                    updated: *updated.as_ref(),
+                    size,
+                    children: vec![],
+                    content,
+                }
+            }
+            Manifest::Folder(FolderManifest {
+                author,
+                timestamp,
+                id,
+                version,
+                created,
+                updated,
+                children,
+                ..
+            })
+            | Manifest::Workspace(WorkspaceManifest {
+                author,
+                timestamp,
+                id,
+                version,
+                created,
+                updated,
+                children,
+                ..
+            }) => {
+                let children = children
+                    .into_iter()
+                    .map(|(name, id)| {
+                        vlob_atom::table
+                            .select(vlob_atom::blob)
+                            .filter(vlob_atom::vlob_id.eq((*id).as_ref()))
+                            .order_by(vlob_atom::timestamp.desc())
+                            .first::<Vec<u8>>(conn)
+                            .map(|x| Self::load(conn, key, name.as_ref().to_string(), &x))
+                    })
+                    .collect::<Result<Result<Vec<_>, _>, _>>()??;
+
+                let size = children.iter().map(|x| x.size).sum();
+
+                Self {
+                    name,
+                    author,
+                    timestamp: *timestamp.as_ref(),
+                    id: *id,
+                    version,
+                    created: *created.as_ref(),
+                    updated: *updated.as_ref(),
+                    size,
+                    children,
+                    content: String::new(),
+                }
+            }
+            _ => unreachable!(),
+        })
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Workspace(pub Data);
+
+impl Workspace {
+    pub fn dump(&self, path: &str) -> Result<(), Box<dyn Error>> {
+        let data = serde_json::to_string_pretty(self)?;
+        std::fs::write(path, &data)?;
```

Comment:
I guess this code is not finished: the cli is supposed to write all the files/folders from the workspace into the output folder